### ISSUE TITLE
How is the Planetoid Cora dataset split?

### ISSUE THREAD

**Issue by vymao** [pyg-team/pytorch_geometric#1860]
## ❓ Questions & Help

I am not particularly familiar with the Cora dataset, but to my understanding, it is one large graph for node classification. Yet I have seen demonstrations that one can minibatch the processing. How is this done? If you partition the training nodes into minibatches, won't you be eliminating potential connections to nodes outside the minibatch subgraph? 

Also, how is masking incorporated into this?

**Comment by rusty1s** [comment]
Cora is a single graph where the task is to infer node labels solely based on a handful of labeled nodes (semi-supervised training). It is actually pretty small, so there is no need to mini-batch the computation (although one certainly can).

For mini-batching, you have to make some compromises. Most scalable methods do this by sampling their L-hop neighborhood, see [here](https://cs.stanford.edu/people/jure/pubs/graphsage-nips17.pdf). Note that this will also incorporate nodes that are not part of the current mini-batch of nodes.

In the case of sampling subgraphs, you will lose out on information of nodes that are not in the current mini-batch. However, in practice, this seems to work fine though (see [here](https://arxiv.org/abs/1905.07953) and [here](https://arxiv.org/abs/1907.04931).

Masking works the same as in the full-batch counterparts. You can either just sample training nodes to form mini-batches, or mask out mini-batch nodes which do not belong to the training set.
